src/ui/mentionsview.js

```diff
@@ -71,14 +98,34 @@ export default class MentionsView extends View {
 		const item = this.listView.items.get( indexToGet );
 		item.highlight();
 
+		// Scroll the mentions view to the selected element.
+		if ( !this._isItemVisibleInScrolledArea( item ) ) {
+			this.element.scrollTop = item.element.offsetTop;
+		}
+
 		if ( this.selected ) {
 			this.selected.removeHighlight();
 		}
 
 		this.selected = item;
 	}
 
+	/**
+	 * Triggers "execute" event on selected item.
+	 */
 	executeSelected() {
 		this.selected.fire( 'execute' );
 	}
+
+	// Checks if item is visible in scrolled area.
+	//
+	// The item is considered visible when:
+	// - its top line is inside scrolled rect
+	// - its bottom line is inside scrolled rect (whole item must be visible)
+	_isItemVisibleInScrolledArea( item ) {
+		const isBottomLineVisible = item.element.offsetTop + item.element.clientHeight <= this.element.clientHeight;
+		const isTopLineVisible = item.element.offsetTop >= this.element.scrollTop;
+
+		return isBottomLineVisible && isTopLineVisible;
```

Comment:
We already have tools for that, use

```js
return new Rect( this.element ).contains( new Rect( item.element ) );
```

instead.

Reply:
Nice one - I din't know that we had such thing :)